src/components/logs/logsWebview.ts

```diff
@@ -61,11 +61,65 @@ export class LogsPanel extends WebPanel {
                 </div>
             </div>
             <script>
-              var lastMode = '';
-              var lastRegexp = '';
-              var renderNonce = 0;
+              let renderNonce = 0;
+              let orig = \`${this.content}\`.split('\\n');
 
-              var orig = \`${this.content}\`.split('\\n');
+              const getFilteredContent = (textToAppend) => {   
+                const regexp = document.getElementById('regexp').value;
+                const mode = document.getElementById('mode').value;                         
+                const rawContent = textToAppend ? textToAppend : orig;
+                let content;
+                if (regexp.length > 0 && mode !== 'all') {
+                    const regex = new RegExp(regexp);   
+                    switch (mode) {                        
+                        case 'include':
+                            content = rawContent.filter((line) => regex.test(line));
+                            break;
+                        case 'exclude':
+                            content = rawContent.filter((line) => !regex.test(line));
+                            break;
+                        case 'before':
+                            content = [];
+                            if (!textToAppend) { 
+                                for (const line of orig) {
+                                    if (regex.test(line)) {
+                                        break;
+                                    }
+                                    content.push(line);
+                                }
+                            }
+                            break;
+                        case 'after':
+                            if (textToAppend) {
+                                content = rawContent;
+                            } else {
+                                const i = orig.findIndex((line) => {
+                                    return regex.test(line)
+                                });
+                                content = orig.slice(i+1);
+                            }                           
+                            break;
+                        default:
+                            content = []
+                            break;
+                    }
+                } else {
+                    content = rawContent;
+                }
+                
+                return content;
+              };
+
+              const getBeautifiedContent = (contentAsArray, ix, end) => {
+                if (ix && end) {
+                    contentAsArray = content.slice(ix, end);
```

Comment:
Is it really your intent to overwrite the input here?  So if the indices are passed then it ignores the passed  `contentArray` altogether?  Feels like we maybe have two distinct functions here e.g. `beautifyContentLineRange(ix, end)` (works on global `content`) and `beautifyLines(lines)` (works on the passed-in array of lines), where the first one performs the slice then calls down to the second.

Reply:
I seriously missed that one. It still worked bc the eval is triggered when a new regex or a new filter mode is picked and it works with the full content ... good catch :+1: fixed that. I also split the function to make the code more readable.